### PR TITLE
Versioning and warning wave related changes for MS.CA.NetAnalyzers pa…

### PR DESCRIPTION
**Pull request by mavasani**
…ckage

1. Update the repo README to mention `Microsoft.CodeAnalysis.NetAnalyzers` package as the primary package for this repo
2. Update `Microsoft.CodeAnalysis.NetAnalyzers` package and assembly versions to `5.0.0`.
3. Update `Microsoft.CodeAnalysis.NetAnalyzers` shipped analyzer release to `5.0` and move all the current CA rules in the package under this release.
4. Updates to GlobalAnalyzerConfig generation tool:
   1. This post build tool now generates global editorconfig files named `AnalysisLevel_x_y_z.editorconfig`, for each shipped release `x.y.z` in the AnalyzerReleases.shipped.md file. For example, for a shipped release with value `5`, it generates `AnalysisLevel_5.editorconfig`. This file is currently emtpy, but adding new analyzers to the repo after this change will lead this file to have entries `dotnet_diagnostic.RuleId.severity = none` (I verified this manually).
   image
   2. The targets file generated by this tool now ensures that correct global editorconfig is passed down to the compiler based on MSBuild property `AnalysisLevel`. For example, for value `5` or `5.1.1` for this property, `AnalysisLevel_5.editorconfig` or `AnalysisLevel_5_1_1.editorconfig` will be passed down to the compiler - I verified this manually by creating a project referencing the locally built NetAnalyzers NuGet package and then specifying `AnalysisLevel = 5` in the project file. Manually adding any new `dotnet_diagnostic.RuleId.severity` entries to this file causes both the IDE and command line builds to respect these configurations.
   image
   image
5. Minor refactorings to some security rules to ensure the release tracking analyzer can track the rule IDs for supported descriptors
6. Minor fix to rules missing documentation file generation to enable it by default for local builds triggered with `/t:pack`. Skip it for CI/signed builds for improved CI throughput.

### CODE REVIEW

File: src/Tools/GenerateDocumentationAndConfigFiles/Program.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ public static int Main(string[] args)
                 containsPortedFxCopRules = false;
             }
 
-            if (!bool.TryParse(args[15], out var generateAnalyzerRulesMissingDocumentationFile))
+            if (!bool.TryParse(args[16], out var generateAnalyzerRulesMissingDocumentationFile))
```
Comment (mavasani) [inline src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:51]: This was a typo, we were using the wrong arg index.

File: eng/Versions.props [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <Project>
   <PropertyGroup>
     <VersionPrefix>3.3.1</VersionPrefix>
+    <NetAnalyzersVersionPrefix>5.0.0</NetAnalyzersVersionPrefix>
     <AnalyzerUtilitiesVersionPrefix>$(VersionPrefix)</AnalyzerUtilitiesVersionPrefix>
     <PreReleaseVersionLabel>beta1</PreReleaseVersionLabel>
```
Comment (jmarolf) [inline eng/Versions.props:6]: Can we have a `PreReleaseVersionLabel` for the `NetAnalyzers` package that is different than the other packages in this repo? For example I would like to lable the current package in master `rc1` as that is the .NET 5 release it is going into

**Comment by jmarolf** [comment]
@mavasani please take a look at https://github.com/dotnet/sdk/pull/12898. I am thinking of adding `EffectiveAnalysisLevel` to handle 'none', 'latest', and 'preview'

File: eng/Versions.props (hunk 2 above)
Comment (mavasani) [reply eng/Versions.props:6]: Yep, let me push a follow-up commit for it.

**Comment by mavasani** [comment]
Thanks @jmarolf. The next NetAnalyzers package should be `5.0.0-rc1`

**Comment by mavasani** [comment]
@jmarolf - https://dev.azure.com/dnceng/public/_packaging?_a=package&feed=dotnet5&package=Microsoft.CodeAnalysis.NetAnalyzers&protocolType=NuGet&version=5.0.0-rc1.20413.2&view=versions

**Pull request merged**